packages/shacl-middleware/index.ts

```diff
@@ -86,15 +101,21 @@ export const shaclMiddleware = ({ getTargetNode, loadResource, loadResourcesType
     return res.status(400).send(response)
   }
 
-  // Load data from linked instances to be able to validate their type
-  const classProperties = clownface({ dataset: shapes })
-    .out(sh.property)
-    .has(sh.class)
-    .out(sh.path)
-  const linkedInstancesIds = resource.out(classProperties).terms.filter(r => r.termType !== 'BlankNode')
-  const linkedInstancesQuads = await loadResourcesTypes(linkedInstancesIds)
-
-  const dataset = $rdf.dataset([...resource.dataset, ...linkedInstancesQuads])
+  let dataset: DatasetCore
+  if (disableShClass) {
+    clownface({ dataset: shapes }).has(sh.class).addOut(sh.deactivated, true)
+
+    dataset = resource.dataset
```

Comment:
Only one hawkish change here. The validation of `sh:class` constraint is problematic in a request context because it's a closed world slice.

In typical cases, there is a root resource and shape with such properties. You will find below how the types for relevant resources is loaded to satisfy the `sh:class` constraints.

Here, however, it would need to be more sophisticated because it's not right away known what is the target for any given property.

Thus, I have added an option to deactivate those properties in the import validation